tap_mysql/__init__.py

```diff
@@ -78,15 +100,11 @@ def binlog_stream_requires_historical(catalog_entry, state):
                                   catalog_entry.tap_stream_id,
                                   'log_pos')
 
-    max_pk_values = singer.get_bookmark(state,
-                                        catalog_entry.tap_stream_id,
-                                        'max_pk_values')
-
-    last_pk_fetched = singer.get_bookmark(state,
-                                          catalog_entry.tap_stream_id,
-                                          'last_pk_fetched')
+    initial_full_table_complete = singer.get_bookmark(state,
```

Comment:
Rather than inspecting max/last PK fetched state, just look at whether the one-off initial-binlog-sync was recorded as completed in the state. It's all-or-nothing now.